modules/concurrency/tests/channel_tests.cpp

```diff
@@ -0,0 +1,119 @@
+//=================================================================================================
+// Copyright (C) 2023-2025 HEPHAESTUS Contributors
+//=================================================================================================
+
+#include <exec/when_any.hpp>
+#include <gtest/gtest.h>
+#include <stdexec/execution.hpp>
+
+#include "hephaestus/concurrency/channel.h"
+
+namespace heph::concurrency {
+TEST(Channel, SendRecv) {
+  Channel<int, 1> c;
+
+  EXPECT_EQ(c.size(), 0);
+
+  {
+    auto [value] = *stdexec::sync_wait(stdexec::when_all(c.getValue(), c.setValue(1) | stdexec::then([&]() {
```

Comment:
how can you guarantee that? couldn't be that `c.getValue()` runs before the value is set?